src/main/java/sirius/biz/ide/Scripting.java

```diff
@@ -0,0 +1,206 @@
+        }
+
+        String jobNumber = Hasher.md5()
+                                 .hash(script)
+                                 .hash(targetNode)
+                                 .hash(LocalDateTime.now().toString())
+                                 .toHexString()
+                                 .substring(0, 6);
+
+        // For audit and tracing purposes we also log this to into the system logs...
+        Cluster.LOG.INFO("Executing administrative script %s for %s on '%s':%n%s",
+                         jobNumber,
+                         UserContext.getCurrentUser().getUserName(),
+                         targetNode,
+                         script);
+
+        logInTranscript(jobNumber,
+                        Strings.apply("Submitting %s on %s for user %s",
+                                      jobNumber,
+                                      CallContext.getNodeName(),
+                                      UserContext.getCurrentUser().getUserName()));
+
+        interconnect.dispatch(getName(),
+                              new JSONObject().fluentPut(TASK_TYPE, TASK_TYPE_EXEC)
+                                              .fluentPut(TASK_SCRIPT, script)
+                                              .fluentPut(TASK_NODE, targetNode)
+                                              .fluentPut(TASK_JOB, jobNumber));
+
+        return jobNumber;
+    }
+
+    @Override
+    public void handleEvent(JSONObject event) {
+        if (TASK_TYPE_MSG.equals(event.getString(TASK_TYPE))) {
+            handleMessageTask(event);
+        } else if (TASK_TYPE_EXEC.equals(event.getString(TASK_TYPE))) {
+            handleExecTask(event);
+        }
+    }
+
+    private void handleMessageTask(JSONObject event) {
+        synchronized (messages) {
+            messages.add(new TranscriptMessage(event.getString(TASK_NODE),
+                                               event.getString(TASK_JOB),
+                                               event.getLong(TASK_TIMESTAMP),
+                                               event.getString(TASK_MESSAGE)));
+            if (messages.size() > MAX_MESSAGES) {
+                messages.remove(0);
+            }
+        }
+    }
+
+    /**
+     * Provides access to the last 250 messages of the transcript.
+     *
+     * @return the last message which were recorded in the transcript
+     */
+    public List<TranscriptMessage> getMessages() {
+        synchronized (messages) {
+            return new ArrayList<>(messages);
+        }
+    }
+
+    private void handleExecTask(JSONObject event) {
+        String nodeName = event.getString(TASK_NODE);
+        String jobNumber = event.getString(TASK_JOB);
+        if (!ALL_NODES.equals(nodeName) && !Strings.areEqual(CallContext.getNodeName(), nodeName)) {
+            return;
+        }
+
+        Watch watch = Watch.start();
+        logInTranscript(jobNumber, Strings.apply("Executing %s on %s", jobNumber, CallContext.getNodeName()));
```

Comment:
Does this duplicate the jobnumber (as in is the jobNumber param printed anywhere in the ui?)? sometimes it is only transferred as a message parameter, other times it is also written in the logged message.